test/common/router/config_impl_test.cc

```diff
@@ -2540,6 +2600,24 @@ TEST(RouteMatcherTest, WeightedClustersMissingClusterList) {
                EnvoyException);
 }
 
+TEST(RouteMatcherTest, WeightedClustersInvalidTotalWeight) {
+  std::string yaml = R"EOF(
+virtual_hosts:
+  - name: www2
+    domains: ["www.lyft.com"]
+    routes:
+      - match: { prefix: "/" }
+        route:
+          weighted_clusters:
+            total_weight: 0
+  )EOF";
+
+  NiceMock<Runtime::MockLoader> runtime;
+  NiceMock<Upstream::MockClusterManager> cm;
+  EXPECT_THROW_WITH_REGEX(ConfigImpl(parseRouteConfigurationFromV2Yaml(yaml), runtime, cm, true),
```

Comment:
Any reason why you are doing regext match instead of complete match with `EXPECT_THROW_WITH_MESSAGE`